### PR TITLE
BUG giving modulo operator back to Int64Index

### PR DESCRIPTION
**Pull request by nbonnotte**
Fix issue #9244

### CODE REVIEW

File: pandas/tests/test_index.py [hunk 1]
```diff
@@ -1304,6 +1304,12 @@ def test_symmetric_diff(self):
         self.assertTrue(tm.equalContents(result, expected))
         self.assertEqual(result.name, 'new_name')
 
+    def test_modulo(self):
+        # GH 9244
+        intidx = Index(range(10))
+        expected = Int64Index([0, 1, 0, 1, 0, 1, 0, 1, 0, 1], dtype='int64')
+        self.assert_index_equal(intidx % 2, expected)
+
```
Comment (jreback) [inline pandas/tests/test_index.py]: you are adding this test in a generic section, so it is not doing what you think. instead define in the ``TestNumeric`` section and use ``.create_index``. As I think this works for more index types.

File: doc/source/whatsnew/v0.18.0.txt [hunk 2]
```diff
@@ -502,4 +502,6 @@ Bug Fixes
 
 - Bug in ``.to_csv`` ignoring formatting parameters ``decimal``, ``na_rep``, ``float_format`` for float indexes (:issue:`11553`)
 
+- Bug in ``Int64Index`` preventing the use of the modulo operator (:issue:`9244`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: this is very narrow what you are saying. pls test & expand (this should work for numeric types)

File: pandas/tests/test_index.py (hunk 1 above)
Comment (nbonnotte) [inline pandas/tests/test_index.py]: You're right, it works for `Float64Index` as well

**Comment by nbonnotte** [comment]
@jreback you were right, this also allows the use of the modulo operator for `Float64Index`

I've updated the tests and the what's new

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/4625d4d889842041b70f11ae16c0757590fef0d8

thanks!

**Pull request closed without merging**